spark-connector/spark-connector/src/test/java/com/datastrato/gravitino/spark/connector/TestSparkTransformConverter.java

```diff
@@ -40,83 +41,105 @@ void init() {
     initSparkToGravitinoTransformMap();
   }
 
-  @Test
-  void testPartition() {
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void testPartition(boolean supportsBucketPartition) {
+    SparkTransformConverter sparkTransformConverter =
+        new SparkTransformConverter(supportsBucketPartition);
+
     sparkToGravitinoPartitionTransformMaps.forEach(
         (sparkTransform, gravitinoTransform) -> {
           Transform[] gravitinoPartitionings =
-              SparkTransformConverter.toGravitinoPartitionings(
+              sparkTransformConverter.toGravitinoPartitionings(
                   new org.apache.spark.sql.connector.expressions.Transform[] {sparkTransform});
-          Assertions.assertTrue(
-              gravitinoPartitionings != null && gravitinoPartitionings.length == 1);
-          Assertions.assertEquals(gravitinoTransform, gravitinoPartitionings[0]);
-        });
-
-    sparkToGravitinoPartitionTransformMaps.forEach(
-        (sparkTransform, gravitinoTransform) -> {
-          org.apache.spark.sql.connector.expressions.Transform[] sparkTransforms =
-              SparkTransformConverter.toSparkTransform(
-                  new Transform[] {gravitinoTransform}, null, null);
-          Assertions.assertTrue(sparkTransforms.length == 1);
-          Assertions.assertEquals(sparkTransform, sparkTransforms[0]);
+          if (sparkTransform instanceof BucketTransform && !supportsBucketPartition) {
+            Assertions.assertTrue(
+                gravitinoPartitionings != null && gravitinoPartitionings.length == 0);
+          } else {
+            Assertions.assertTrue(
+                gravitinoPartitionings != null && gravitinoPartitionings.length == 1);
+            Assertions.assertEquals(gravitinoTransform.name(), gravitinoPartitionings[0].name());
```

Comment:
should check `Assertions.assertEquals(gravitinoTransform, gravitinoPartitionings[0]);`

Reply:
fixed.